docs/meshstack.configuration.md

```diff
@@ -134,11 +135,203 @@ When a user is invited to a customer there are several configurations in order t
+      , identity-lookup : Optional IdentityLookup
       }
 }
 ```
 
+#### User Setup Steps
+
+As explained certain steps can be used in order to setup initial user. To use the Azure lookup functionality a principal must be configured like so:
+
+<!--snippet:mesh.meshfed.identity.azure.creds#type-->
+
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Dhall Type-->
+```dhall
+let AzureCreds =
+    {-
+      Setting this configuration enables the use of an AAD as a user lookup source to allow
+      autocomplete of user information when adding new users to customers.
+
+        aad-tenant:
+            The active directory tenant. Its either a UID of the AAD or its domain
+            like devmeshcloud.onmicrosoft.com
+
+        client-id:
+            The client id of the service principal
+
+        client-secret:
+            The credentials of the service principal
+
+        user-lookup-strategy:
+            The lookupstrategy which is used in order to find the user. Use either
+            UserByUsernameLookupStrategy or UserByMailLookupStrategy. The UserByMailLookupStrategy
+            uses the users euid and uses it as an E-Mail address for AAD lookup. The
+            UserByUsernameLookupStrategy assumes the euid is an userPrincipalName for AAD lookup.
+    -}
+      { aad-tenant : Text
+      , client-id : Text
+      , client-secret : Secret.Type
+      , user-lookup-strategy : AzureLookupStrategy
+      }
+```
+<!--Example-->
+```dhall
+let example
+    : AzureCreds
+    = { aad-tenant = "devmeshcloud.onmicrosoft.com"
+      , client-id = "f112f31-248a-4461-1269-0f13164acb95"
+      , client-secret = Secret.fromAnsible "client_secret"
+      , user-lookup-strategy =
+          AzureLookupStrategy.UserByMailLookupStrategy
+      }
+```
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+If this initial config if present you can decide to setup the following optional user identity steps:
+
+<!--snippet:mesh.meshfed.identity.azure.guest#type-->
+
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Dhall Type-->
+```dhall
+let AzureGuestDetection =
+    {-
+      When adding/inviting a new user to a customer a custom attribute property from the AAD data
+      can be used to determine if it is a guest user.
+      Attention: This check is only performed on the first attempt when a user is added/invited
+      to a customer. If this check is configured after some users were initially added to a
+      customer they are not detected as guest users anymore.
+
+        guest-property:
+            The AAD's custom attribute which is checked against the guest-value.
+
+        guest-value:
+            If the AAD custom attribute matches this value the user is considered to be a guest.
+    -}
+      { guest-property : Text, guest-value : Text }
+```
+<!--Example-->
+```dhall
+let example
+    : AzureGuestDetection
+    = { guest-property = "guest-property", guest-value = "is-guest" }
+```
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+<!--snippet:mesh.meshfed.identity.azure.euid#type-->
```

Comment:
f: This part about identity lookup setup is duplicating https://docs.meshcloud.io/docs/meshstack.configuration.html#configure-for-azure-aad

d: As this is now getting a lot of material for lookup and user-invitation setup, it could make sense to give it it's own page.

Reply:
updated and unified the configs. About the question for the dedicated page... yeah. But I originally did not want to do that because then we have this one single page outstanding there... I am just not sure about the overall structure of the docs in that area. Maybe we should think about if we can find a better/more organized structure in general and then push/sort all the pages once.